src/parseobject.cpp

```diff
@@ -453,8 +460,21 @@ void ParseObject::addAtomNewsIntoBase(NewsItemStruct *newsItem)
     if (isDuplicate) break;
   }
 
-  // if duplicates not found, add news into base
-  if (!isDuplicate) {
+  // Verify old news before a date to avoid adding them to base
+  bool isOld = false;
+  QDateTime pubDate_ = QDateTime::fromString(newsItem->updated, "yyyy-MM-ddTHH:mm:ss");
+  QDateTime avoidedDate_ = QDateTime(mainApp->mainWindow()->avoidedOldNewsDate_);
+  if (!addSingleNewsAnyDate_) {      //
+    if (avoidedOldSingleNews_ ) {     // avoid adding old single news
+      if (QDateTime(avoidedOldSingleNewsDate_) > pubDate_)
+        isOld = true;
+      } else if (mainApp->mainWindow()->avoidOldNews_ && avoidedDate_ > pubDate_) {   // avoid adding old news
+        isOld = true;
+      }
+   }
+
+  // if duplicates not found and is old news, add them into base
+  if (isDuplicate == false and isOld == false) {
```

Comment:
The appveyor code style checker is complaining here because it expects the condition checks to be contained within parens. How about using the original code style? This line could be changed to `if (!isDuplicate and !isOld) {`. The same idea applies to other pieces of code where you compare to `true` or `false`.

Reply:
Also, it looks like the `and` operator is not known to MSVC so compilation fails on Windows. How about using `&&` instead?

Reply:
I updated as you said to:   **`if (!isDuplicate && !isOld)`**
Waiting for the result of the AppVeyor check.

Reply:
Nice. Builds fine now on all systems.